### PR TITLE
T9

### PR DESCRIPTION
**Pull request by RaymondBibby**
Add completed end point GET /api/articles/:article_id/comments:

Happy path:
200: serves response correctly as expected

Sad Path:
400 bad article_id e.g., banana SQL 22P02 error
404 valid article_id but no such resource exists
404 misspelled articles

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -230,12 +230,66 @@ describe("GET: /api/articles", () => {
...
 			.expect(404)
 			.then(({ body }) => {
-				expect(body.msg).toBe("Invalid input, no such end point exists");
+				expect(body.msg).toBe('Invalid input, no such end point exists');
 			});
 	});
 });
+
+describe('GET: /api/articles:article_id/comments', () => {
+	describe('HAPPPY  path', () => {
```
Comment (alexdevdavis) [inline __tests__/app.test.js:244]: typo

File: __tests__/app.test.js [hunk 2]
```diff
@@ -230,12 +230,66 @@ describe("GET: /api/articles", () => {
...
+		});
+		it("Status 404: responds with a 404 and an appropriate message when a valid end point is provided but the resource doesn't exist", () => {
+			return request(app)
+				.get('/api/articles/999/comments')
+				.expect(404)
+				.then(({ body }) => {
+					expect(body.msg).toBe('No article found for article_id 999');
+				});
+		});
+		it('Status 404: resposnds with a 404 bad request and an appropriate message when an invalid end point is provided, e.g., misspelling', () => {
```
Comment (alexdevdavis) [inline __tests__/app.test.js:286]: typo

File: __tests__/app.test.js [hunk 3]
```diff
@@ -230,12 +230,66 @@ describe("GET: /api/articles", () => {
...
+			return request(app)
+				.get('/api/articlles/1/comments')
+				.expect(404)
+				.then(({ body }) => {
+					expect(body.msg).toBe('Invalid input, no such end point exists');
+				});
+		});
```
Comment (alexdevdavis) [inline __tests__/app.test.js:293]: consider your desired behaviour (and write a test) for when a valid endpoint is provided, but there aren't any comments yet

File: app.js [hunk 4]
```diff
@@ -10,38 +10,42 @@ const {
...
+app.get('/api/articles/:article_id/comments', getCommentsByArticleId);
+
+app.all('/api/*', catchAll);
+
+// Error handling block
 
-/////ERROR HANDLING BLOCK///////
 app.use((err, req, res, next) => {
-	console.log("Entering Error handling block", err);
+	// console.log('Entering Error handling block', err);
```
Comment (alexdevdavis) [inline app.js]: it's fine to have this console log as part of your code, but decide whether you want it or not ... commented out code isn't the pro look you're evidently capable of

File: models/news_models.js [hunk 5]
```diff
@@ -43,17 +43,31 @@ exports.fetchArticles = () => {
...
 		}
 	);
 };
+
+exports.fetchCommentsByArticleId = async ({ article_id }) => {
+	const result = await db.query('SELECT * FROM comments WHERE article_id=$1', [
+		article_id,
+	]);
+
+	if (result.rows.length === 0) {
```
Comment (alexdevdavis) [inline models/news_models.js]: this condition would return true if article is found but with no comments yet

**Review by alexdevdavis — CHANGES_REQUESTED** [review]
code looks good overall ... you just need to ensure that you account for the situation of a valid article with no comments

File: app.js (hunk 4 above)
Comment (RaymondBibby) [reply app.js]: I'll leave it uncommented until its ready to be hosted and all testing is completed :)

File: __tests__/app.test.js (hunk 1 above)
Comment (RaymondBibby) [reply __tests__/app.test.js:244]: Resolved

File: __tests__/app.test.js (hunk 2 above)
Comment (RaymondBibby) [reply __tests__/app.test.js:286]: Resolved

File: models/news_models.js (hunk 5 above)
Comment (RaymondBibby) [reply models/news_models.js]: resolved :)

File: __tests__/app.test.js (hunk 3 above)
Comment (RaymondBibby) [reply __tests__/app.test.js:293]: Took a little longer than expected, but got bot cases working :)

**Pull request merged**